docs/design/yarn-v1.md

```diff
@@ -0,0 +1,178 @@
+limitation should be documented.
+
+### SBOM Generation
+The Yarn 1.x CLI doesn't provide enough information to generate an SBOM, so the yarn.lock and package.json files will need to
+be parsed in order to gather the necessary data. The code used to process these files will need to be
+[imported from or inspired by Cachito](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py)
+and refactored/improved.
+
+#### Key Points
+ - The yarn.lock file can be parsed with [pyarn](https://github.com/containerbuildsystem/pyarn) like in cachito
+ - Workspaces will need to be [processed separately](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L95) from package.json since they do not appear in the yarn.lock file
+ - Dev dependencies can be [determined similiarly](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L140) to cachito
+ - We will need to ensure that there are no collisions in the names of the archive files being added to the offline mirror
+
+#### PURL Examples
+[NPM PURL specification](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#npm)
+
+##### Registry Package
+```txt
+pkg:npm/@optional-scope/package-name@1.0.0
+```
+
+##### URL Package
+```txt
+pkg:npm/package-name@1.0.0?checksum=sha512:checksum&download_url=https://example.com/package-name/0101010101010101010101010101010101010101.tar.gz
+```
+
+##### Git Package
+
+```txt
+pkg:npm/package-name@1.0.0?vcs_url=git%2Bhttps://example.com/namespace/package-name.git%0101010101010101010101010101010101010101
+```
+
+##### Workspace/File Package
+
+```txt
+pkg:npm/package-name@1.0.0?vcs_url=git%2Bhttps://example.com/namespace/package-name.git%0101010101010101010101010101010101010101#subpath
+```
+
+### Avoiding Arbitrary Code Execution
+Yarn 1.x appears to honor the `--ignore-scripts` option for the install command. It can also be configured via environment variable.
+Used https://github.com/chmeliik/js-lifecycle-scripts/tree/yarn to verify.
+
+### Prefetch
+The offline mirror in the request output directory will be populated by running the following Yarn command in the project root:
+
+    yarn install --non-interactive --frozen-lockfile --disable-pnp --no-default-rc --ignore-engines
+
+Command: [yarn install](https://classic.yarnpkg.com/en/docs/cli/install)
+
+Options:
+ - [non-interactive](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-non-interactive): Disable interactive prompts
+ - [frozen-lockfile](https://classic.yarnpkg.com/en/docs/cli/install#toc-yarn-install-frozen-lockfile): Fail if yarn.lock updates are needed
+ - [disable-pnp](https://github.com/yarnpkg/yarn/blob/7cafa512a777048ce0b666080a24e80aae3d66a9/src/cli/index.js#L84): Disable Plug'n'Play installation
```

Comment:
I noticed that when we set this flag, Yarn removes the `installConfig: pnp: true` key from the package.json file. Since the idea is to execute the commands in a copy of the source files, we're safe against unwanted modifications.

But I still wonder how we should treat cases where this flag is set. The prefetched content would not be in the right format for a pnp offline install, right? Should we just fail the request with an unsupported error?

Reply:
> is pnp the type of installation in Yarn v1 where you check in the whole cache (or even node_modules to your repo?

It appears to be a way to avoid generating the node_modules directory for performance purposes. A pnp project will still try to fetch all of the dependencies from the cache/offline-mirror/internet

Reply:
> Should we just fail the request with an unsupported error?

I'm leaning this way now because I tested with a slightly different project this time and saw an appearance of our old (in this case undocumented) friend, unplugged packages. Apparently this is only possible with pnp enabled.

Reply:
We have decied to not support PnP, closing the thread.